### DISCUSSION TITLE
Can clippy lints look at regular (not doc) comments?

### DISCUSSION THREAD

**Discussion by A-Walrus** [rust-lang/rust-clippy#10505] [General]
I'm trying to develop a lint that detects cases like:
```rust
// This is a regular comment which documents this function
fn foo() {
}
```
And recommends changing it to:
```rust
/// This is a doc comment which documents this function
fn foo() {
}
```

From my understanding regular comments don't appear in the AST so I'm not sure if such a lint is even possible...
Any ideas?

**Comment by xFrednet** [comment]
We have a few lints that check for such comments, you can take a look at: [`clippy::undocumented_unsafe_blocks`](https://rust-lang.github.io/rust-clippy/master/index.html#undocumented_unsafe_blocks). Generally speaking, I think we're careful about such lints, as they can be finicky exactly because they're not stored in the AST. However, I would guess that a lint for the snippets above would be accepted. :)

**Reply by flip1995** [reply]
To get those comments, Clippy must re-lex the source file though. The lints that do this know the exact span they have to re-lex and can limit it to a small part of code. So a lint that needs to re-lex an entire file or even crate to find comments won't be allowed. If the lint can guarantee that only a small part in only few situations have to be re-lexed that is doable.
